### PR TITLE
feat: handle tag delete action

### PR DESCRIPTION
**Pull request by oussama-aitbaziz-dev**
This PR should make `<Tag />` handle onDelete callback

### CODE REVIEW

File: src/ui/component/tag/tag.tsx [hunk 1]
```diff
@@ -1,12 +1,35 @@
...
-    <div className="okp4-dataverse-portal-tag-name">{tagName}</div>
-  </div>
-)
+export const Tag: FC<TagProps> = ({ tagName, onDelete }) => {
+  const { t } = useTranslation('common')
+  const handleDelete = useCallback(() => {
+    if (onDelete) {
+      onDelete()
+    }
+  }, [onDelete])
```
Comment (fredericvilcot) [inline src/ui/component/tag/tag.tsx:17]: This method is unecessary

File: src/ui/component/tag/tag.tsx [hunk 2]
```diff
@@ -1,12 +1,35 @@
...
+
+  return (
+    <div className="okp4-dataverse-portal-tag-main">
+      <div className="okp4-dataverse-portal-tag-name">{tagName}</div>
+      {onDelete && (
+        <div
+          className="okp4-dataverse-portal-tag-icon"
+          onClick={handleDelete}
```
Comment (fredericvilcot) [inline src/ui/component/tag/tag.tsx:25]: You've already made an assertion on the `onDelete` function, so you can give it directly to the event handler.
```suggestion
          onClick={onDelete}
```

**Review by fredericvilcot — CHANGES_REQUESTED** [review]

**Review by lolottetheclash — CHANGES_REQUESTED** [review]

**Review by fredericvilcot — APPROVED** [review]
LGTM! 👍

**Review by lolottetheclash — APPROVED** [review]
Let's go !

**Pull request merged**